### PR TITLE
Added support for SFN (Step Functions)

### PR DESCRIPTION
**Pull request by bwhove**
As requested over here: https://github.com/quarkiverse/quarkus-amazon-services/issues/440, I've attempted to add support for amazon-sfn (Step Functions).

I'm not sure if everything is as it should be, but it compiles and seems te be working. 

I do have some doubts about the naming of the configuration properties prefix. Currently, it's `quarkus.stepfunctions`, e.g.

```
quarkus.stepfunctions.endpoint-override=http://endpoint
```

But maybe it should be aligned with the Amazon Client name?

```
quarkus.sfn.endpoint-override=http://endpoint
```

As far as documentation goes, I've added some generic docs, but will also work on providing some code examples.

### CODE REVIEW

File: sfn/runtime/src/main/java/io/quarkus/amazon/sfn/runtime/SfnBuildTimeConfig.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import io.quarkus.amazon.common.runtime.HasSdkBuildTimeConfig;
+import io.quarkus.amazon.common.runtime.SyncHttpClientBuildTimeConfig;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+
+/**
+ * Amazon SFN build time configuration
+ */
+@ConfigMapping(prefix = "quarkus.stepfunctions")
```
Comment (scrocquesel) [inline sfn/runtime/src/main/java/io/quarkus/amazon/sfn/runtime/SfnBuildTimeConfig.java:14]:
```suggestion
@ConfigMapping(prefix = "quarkus.sfn")
```

File: sfn/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  keywords:
+    - "sfn"
+    - "aws"
+    - "amazon"
+  categories:
+    - "data"
+  guide: https://docs.quarkiverse.io/quarkus-amazon-services/dev/amazon-sfn.html
+  status: "stable"
+  config:
+  - "quarkus.stepfunctions."
```
Comment (scrocquesel) [inline sfn/runtime/src/main/resources/META-INF/quarkus-extension.yaml:14]:
```suggestion
  - "quarkus.sfn."
```

**Review by scrocquesel — CHANGES_REQUESTED** [review]
Thank you for this contribution. For the configuration, the convention is to keep the artifactid from the awssdk. And we should be all good.

**Comment by bwhove** [comment]
Thanks for the review and a happy new year!

Not sure if I understand it correctly, but I think there is now a discrepancy between the properties as expected via `SfnBuildTimeConfig` and `SfnConfig` (where we determined that `@ConfigMapping(prefix = "quarkus.sfn")` ), and the config properties that are set via the `AbstractDevServicesLocalStackProcessor`. I think this currently only applies to the `stepfunctions` service.

So, we have something in our application.properties like:

```properties
quarkus.sfn.endpoint-override=http://endpoint
quarkus.sfn.aws.region=us-east-1
...
```

But at the same time, we also have expexted configuration like below for dev services configuration in the `AbstractDevServicesLocalStackProcessor`, since the `ENDPOINT_OVERRIDE`, `AWS_REGION`, `AWS_CREDENTIALS_TYPE`, `AWS_CREDENTIALS_STATIC_PROVIDER_ACCESS_KEY_ID` and `AWS_CREDENTIALS_STATIC_PROVIDER_SECRET_ACCESS_KEY` are resolved with the `EnabledService#getName` value, which returns `stepfunctions` in this case.

```properties
quarkus.stepfunctions.endpoint-override=http://endpoint
quarkus.stepfunctions.aws.region=us-east-1
...
```

I'm not sure if this would work, right?

We could maybe introduce some utility class to get the property configuration name based on the EnabledService and return `sfn` in case of `LocalStackContainer.Service.STEPFUNCTIONS`. In all other cases, the `EnabledService#getName` should be returned.

```java
public final class LocalStackDevServicesConfigUtil {

    private LocalStackDevServicesConfigUtil() {
    }

    /**
     * Returns the property configuration name for the given {@link LocalStackContainer.EnabledService}.
     * <p>
     * The property configuration name is the name of the service, which is the same as the AWSSDK artifact id.
     * The only exception is the Step Functions service, which is named "sfn" in the AWSSDK and "stepfunctions" in the
     * LocalStack configuration.
     * <p>
     *
     * @param enabledService the LocalStack enabled service
     * @return the property configuration name
     */
    public static String getPropertyConfigurationName(LocalStackContainer.EnabledService enabledService) {
        if (enabledService == LocalStackContainer.Service.STEPFUNCTIONS)
            return "sfn";
        return enabledService.getName();
    }

}
```

In the `AbstractDevServicesLocalStackProcessor` we could then use the return value of `LocalStackDevServicesConfigUtil#getPropertyConfigurationName(enabledService)` instead of the current return value of `enabledService#getName` when setting up the config map.

What's your view on this?

**Comment by scrocquesel** [comment]
> Thanks for the review and a happy new year!
> 
> Not sure if I understand it correctly, but I think there is now a discrepancy between the properties as expected via `SfnBuildTimeConfig` and `SfnConfig` (where we determined that `@ConfigMapping(prefix = "quarkus.sfn")` ), and the config properties that are set via the `AbstractDevServicesLocalStackProcessor`. I think this currently only applies to the `stepfunctions` service.
> 
> So, we have something in our application.properties like:
> 
> ```ini
> quarkus.sfn.endpoint-override=http://endpoint
> quarkus.sfn.aws.region=us-east-1
> ...
> ```
> 
> But at the same time, we also have expexted configuration like below for dev services configuration in the `AbstractDevServicesLocalStackProcessor`, since the `ENDPOINT_OVERRIDE`, `AWS_REGION`, `AWS_CREDENTIALS_TYPE`, `AWS_CREDENTIALS_STATIC_PROVIDER_ACCESS_KEY_ID` and `AWS_CREDENTIALS_STATIC_PROVIDER_SECRET_ACCESS_KEY` are resolved with the `EnabledService#getName` value, which returns `stepfunctions` in this case.
> 
> ```ini
> quarkus.stepfunctions.endpoint-override=http://endpoint
> quarkus.stepfunctions.aws.region=us-east-1
> ...
> ```
> 
> I'm not sure if this would work, right?
> 
> We could maybe introduce some utility class to get the property configuration name based on the EnabledService and return `sfn` in case of `LocalStackContainer.Service.STEPFUNCTIONS`. In all other cases, the `EnabledService#getName` should be returned.
> 
> ```java
> public final class LocalStackDevServicesConfigUtil {
> 
>     private LocalStackDevServicesConfigUtil() {
>     }
> 
>     /**
>      * Returns the property configuration name for the given {@link LocalStackContainer.EnabledService}.
>      * <p>
>      * The property configuration name is the name of the service, which is the same as the AWSSDK artifact id.
>      * The only exception is the Step Functions service, which is named "sfn" in the AWSSDK and "stepfunctions" in the
>      * LocalStack configuration.
>      * <p>
>      *
>      * @param enabledService the LocalStack enabled service
>      * @return the property configuration name
>      */
>     public static String getPropertyConfigurationName(LocalStackContainer.EnabledService enabledService) {
>         if (enabledService == LocalStackContainer.Service.STEPFUNCTIONS)
>             return "sfn";
>         return enabledService.getName();
>     }
> 
> }
> ```
> 
> In the `AbstractDevServicesLocalStackProcessor` we could then use the return value of `LocalStackDevServicesConfigUtil#getPropertyConfigurationName(enabledService)` instead of the current return value of `enabledService#getName` when setting up the config map.
> 
> What's your view on this?

Another idea could be to pass the property name to the setup method but it come with far too much modifications for just this case. Let's do it with the helper method.

**Comment by bwhove** [comment]
> > Thanks for the review and a happy new year!
> > Not sure if I understand it correctly, but I think there is now a discrepancy between the properties as expected via `SfnBuildTimeConfig` and `SfnConfig` (where we determined that `@ConfigMapping(prefix = "quarkus.sfn")` ), and the config properties that are set via the `AbstractDevServicesLocalStackProcessor`. I think this currently only applies to the `stepfunctions` service.
> > So, we have something in our application.properties like:
> > ```ini
> > quarkus.sfn.endpoint-override=http://endpoint
> > quarkus.sfn.aws.region=us-east-1
> > ...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But at the same time, we also have expexted configuration like below for dev services configuration in the `AbstractDevServicesLocalStackProcessor`, since the `ENDPOINT_OVERRIDE`, `AWS_REGION`, `AWS_CREDENTIALS_TYPE`, `AWS_CREDENTIALS_STATIC_PROVIDER_ACCESS_KEY_ID` and `AWS_CREDENTIALS_STATIC_PROVIDER_SECRET_ACCESS_KEY` are resolved with the `EnabledService#getName` value, which returns `stepfunctions` in this case.
> > ```ini
> > quarkus.stepfunctions.endpoint-override=http://endpoint
> > quarkus.stepfunctions.aws.region=us-east-1
> > ...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I'm not sure if this would work, right?
> > We could maybe introduce some utility class to get the property configuration name based on the EnabledService and return `sfn` in case of `LocalStackContainer.Service.STEPFUNCTIONS`. In all other cases, the `EnabledService#getName` should be returned.
> > ```java
> > public final class LocalStackDevServicesConfigUtil {
> > 
> >     private LocalStackDevServicesConfigUtil() {
> >     }
> > 
> >     /**
> >      * Returns the property configuration name for the given {@link LocalStackContainer.EnabledService}.
> >      * <p>
> >      * The property configuration name is the name of the service, which is the same as the AWSSDK artifact id.
> >      * The only exception is the Step Functions service, which is named "sfn" in the AWSSDK and "stepfunctions" in the
> >      * LocalStack configuration.
> >      * <p>
> >      *
> >      * @param enabledService the LocalStack enabled service
> >      * @return the property configuration name
> >      */
> >     public static String getPropertyConfigurationName(LocalStackContainer.EnabledService enabledService) {
> >         if (enabledService == LocalStackContainer.Service.STEPFUNCTIONS)
> >             return "sfn";
> >         return enabledService.getName();
> >     }
> > 
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > In the `AbstractDevServicesLocalStackProcessor` we could then use the return value of `LocalStackDevServicesConfigUtil#getPropertyConfigurationName(enabledService)` instead of the current return value of `enabledService#getName` when setting up the config map.
> > What's your view on this?
> 
> Another idea could be to pass the property name to the setup method but it come with far too much modifications for just this case. Let's do it with the helper method.

Alright, I've processed your comments and updated the `AbstractDevServicesLocalStackProcessor` with the helper method.

**Review by scrocquesel — APPROVED** [review]

**Comment by scrocquesel** [comment]
@all-contributors please add @bwhove  for code

**Pull request merged**